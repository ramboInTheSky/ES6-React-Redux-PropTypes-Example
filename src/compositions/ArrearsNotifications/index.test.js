import React from 'react';
import { shallow } from 'enzyme';

import { ArrearsNotifications } from './';

jest.mock('nhh-styles', () => ({
  generateId: () => 'abc123',
  NotificationPanel: props => <div id="NotificationPanel-mock" {...props} />,
  Typography: {
    P: props => <p id="P-mock" {...props} />,
  },
}));

describe('<ArrearsNotifications />', () => {
  let props;
  let el;

  beforeEach(() => {
    props = {
      notifications: [
        {
          dataBddPrefix: 'prefix',
          title: 'notificatin title',
          lines: [{ dataBdd: 'lines', text: 'first line' }, { text: 'second line' }],
          notificationType: 'confirmation',
        },
      ],
      removeNotification: jest.fn(),
    };
    el = shallow(<ArrearsNotifications {...props} />);
  });

  it('should render the page', () => {
    expect(el).toMatchSnapshot();
  });
});
