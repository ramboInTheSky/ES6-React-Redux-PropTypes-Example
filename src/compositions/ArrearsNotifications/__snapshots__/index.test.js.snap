// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrearsNotifications /> should render the page 1`] = `
<React.Fragment>
  <styled.div
    data-bdd="prefix"
    key="abc123"
  >
    <NotificationPanel
      icon="warning"
      show={true}
      title="notificatin title"
    >
      <React.Fragment>
        <P
          key="lines"
        >
          <span
            data-bdd="prefix-lines"
          >
            first line
          </span>
        </P>
        <P
          key="second line"
        >
          <span>
            second line
          </span>
        </P>
      </React.Fragment>
    </NotificationPanel>
  </styled.div>
</React.Fragment>
`;
